lib/stdlib/src/erl_scan.erl

```diff
@@ -93,7 +93,7 @@
 -type text_fun() :: fun((atom(), string()) -> boolean()).
 -type option() :: 'return' | 'return_white_spaces' | 'return_comments'
                 | 'text' | {'reserved_word_fun', resword_fun()}
-                | {'text_fun', text_fun()}.
+                | {'text_fun', text_fun()} | 'checks'.
```

Comment:
This new `checks` option will unfortunately be visible in the documentation, and as far as I know, there is no easy way to avoid that.

Therefore, I think that the option should have a more specific name, such as `ssa_checks` or `internal_compiler_checks`, and that it should be documented but with a note that is experimental and that it can be changed or removed at any time without prior warning. The same goes for the other options introduced in the other modules.

Reply:
Should we have an option that is `{compiler_internal, term()}` and to clearly signal when something is experimental  (with `{compiler_internal, ssa_checks}` used for this particular case)?

Reply:
Yes, that's a good idea.